bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/BigtableHBaseSettings.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.wrappers;
+
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.BIGTABLE_ADMIN_HOST_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.BIGTABLE_BULK_MAX_ROW_KEY_COUNT;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.BIGTABLE_HOST_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.BIGTABLE_PORT_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.BIGTABLE_USE_GCJ_CLIENT;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.INSTANCE_ID_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.PROJECT_ID_KEY;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase.wrappers.classic.BigtableHBaseClassicSettings;
+import com.google.cloud.bigtable.hbase.wrappers.veneer.BigtableHBaseVeneerSettings;
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+
+/** For internal use only - public for technical reasons. */
+@InternalApi("For internal usage only")
+public abstract class BigtableHBaseSettings {
+
+  protected static final Logger LOG = new Logger(BigtableOptionsFactory.class);
+
+  protected final Configuration configuration;
+  private final String projectId;
+  private final String instanceId;
+
+  public static BigtableHBaseSettings create(Configuration configuration) {
+    if (configuration.getBoolean(BIGTABLE_USE_GCJ_CLIENT, false)) {
+      return new BigtableHBaseVeneerSettings(configuration);
+    } else {
+      return new BigtableHBaseClassicSettings(configuration);
+    }
+  }
+
+  public BigtableHBaseSettings(Configuration configuration) {
+    this.configuration = configuration;
```

Comment:
If we are cloning the instance of configuration at BigtableHBaseSettings level, then we have to keep the original configuration in classic & veneer.

Because there is possibility that a user could override the Credentials using `BigtableExtendedConfiguration`, I hope having configuration at parent & child is fine.

Reply:
Ah, good point, please add a comment so it doesn't get changed in the future